### PR TITLE
Make CAPA tooling work on arm64

### LINKED ISSUE #2930: Make CAPA tooling work on arm64
We should remove some hard-coded `amd64` to allow dev tooling to work on Darwin/arm64 following cluster-api.

See kubernetes-sigs/cluster-api#5547

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
- Make CAPA tooling work on arm64

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2930 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by Ankitasw** [comment]
Thanks for the PR @Prajyot-Parab 🎉 
Was it possible for you to test it in darwin/arm64 machine? Unfortunately I don't have one otherwise would have tested.

**Comment by Prajyot-Parab** [comment]
> Thanks for the PR @Prajyot-Parab 🎉
> Was it possible for you to test it in darwin/arm64 machine? Unfortunately I don't have one otherwise would have tested.

@Ankitasw I don't have access to a `darwin/arm64` machine at the moment, will update if I get hold of one by EOD today.

**Comment by Ankitasw** [comment]
Thank you for verifying @Prajyot-Parab.
/lgtm
cc @sedefsavas @richardcase for approval.

File: Makefile [hunk 1]
```diff
@@ -361,7 +361,7 @@ managers: ## Alias for manager-aws-infrastructure
 
 .PHONY: manager-aws-infrastructure
 manager-aws-infrastructure: ## Build manager binary
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "${LDFLAGS} -extldflags '-static'" -o $(BIN_DIR)/manager .
+	go build -ldflags "${LDFLAGS} -extldflags '-static'" -o $(BIN_DIR)/manager .
```
Comment (richardcase) [inline Makefile:364]: We have lost the `CGO_ENABLED=0` (i.e. disabling cgo). Was this intended?

File: hack/ensure-kind.sh [hunk 2]
```diff
@@ -20,18 +20,20 @@ set -o pipefail
 
 GOPATH_BIN="$(go env GOPATH)/bin/"
 MINIMUM_KIND_VERSION=v0.12.0
+goarch="$(go env GOARCH)"
```
Comment (richardcase) [inline hack/ensure-kind.sh:23]: nit: existing variables are all upper case

Comment (Prajyot-Parab) [reply hack/ensure-kind.sh]: Fixed.

File: Makefile (hunk 1 above)
Comment (Prajyot-Parab) [reply Makefile]: Fixed.

File: Makefile [hunk 3]
```diff
@@ -356,7 +356,7 @@ managers: ## Alias for manager-aws-infrastructure
 
 .PHONY: manager-aws-infrastructure
 manager-aws-infrastructure: ## Build manager binary
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "${LDFLAGS} -extldflags '-static'" -o $(BIN_DIR)/manager .
+	CGO_ENABLED=0 go build -ldflags "${LDFLAGS} -extldflags '-static'" -o $(BIN_DIR)/manager .
```
Comment (richardcase) [inline Makefile:359]: Should we also be adding this:

```
GOOS=linux GOARCH=${ARCH}
```

Or something similar. Do we class the controller manager binary dev tooling? Probably not.

@sedefsavas @pydctw - wdyt?

Comment (Prajyot-Parab) [reply Makefile:359]:
```
OS=$(go env GOOS)
GOOS=${OS} GOARCH=${ARCH}
```
@richardcase something like this should also work I believe

Comment (richardcase) [reply Makefile:359]: That would work 👍 

(the reason in my example for hardcoding linux (`GOOS=linux`) is that we only support running the manager on linux so should we allow the manager to be built for windows or mac?)

Comment (Prajyot-Parab) [reply Makefile:359]: @richardcase In that case it would make sense to have it as `GOOS=linux GOARCH=${ARCH}` since the support is only for linux.

Comment (richardcase) [reply Makefile:359]: That was my thought, but for `clusterawsadm` its ok to compile for windows/mac/linux etc

Comment (Prajyot-Parab) [reply Makefile]: Done.

**Comment by Prajyot-Parab** [comment]
@Ankitasw I think we are good to move forward with this one, let me know incase of any issues.

**Comment by richardcase** [comment]
/approve

@Ankitasw - will leave the lgtm to you if you are happy

**Pull request merged**